java/vector/src/main/java/org/apache/arrow/vector/BaseNullableFixedWidthVector.java

```diff
@@ -0,0 +1,627 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+import io.netty.buffer.ArrowBuf;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.arrow.memory.OutOfMemoryException;
+import org.apache.arrow.memory.BaseAllocator;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.schema.ArrowFieldNode;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.CallBack;
+import org.apache.arrow.vector.util.OversizedAllocationException;
+import org.apache.arrow.vector.util.TransferPair;
+
+public abstract class BaseNullableFixedWidthVector extends BaseValueVector
+        implements FixedWidthVector, FieldVector {
+   private final byte typeWidth;
+
+   private int valueAllocationSizeInBytes;
+   private int validityAllocationSizeInBytes;
+
+   protected final Field field;
+   private int allocationMonitor;
+   protected ArrowBuf validityBuffer;
+   protected ArrowBuf valueBuffer;
+   protected int valueCount;
+
+   public BaseNullableFixedWidthVector(final String name, final BufferAllocator allocator,
+                                       FieldType fieldType, final byte typeWidth) {
+      super(name, allocator);
+      this.typeWidth = typeWidth;
+      valueAllocationSizeInBytes = INITIAL_VALUE_ALLOCATION * typeWidth;
+      validityAllocationSizeInBytes = getSizeFromCount(INITIAL_VALUE_ALLOCATION);
+      field = new Field(name, fieldType, null);
+      valueCount = 0;
+      allocationMonitor = 0;
+      validityBuffer = allocator.getEmpty();
+      valueBuffer = allocator.getEmpty();
+   }
+
+   /* TODO:
+    * Determine how writerIndex and readerIndex need to be used. The
+    * old implementation seems to be using them but not sure how they impact the
+    * operations on vectors -- especially value retrieval and set.
+    * It also seems like that writerIndex in old implementation is useful only
+    * if setValueCount() has been done else it is always 0. This again implies
+    * that it is probably no effect on general purpose vector operations.
+    */
+
+   protected abstract org.slf4j.Logger getLogger();
+
+   @Override
+   public Mutator getMutator() {
```

Comment:
If we are going to remove Mutator/Accessor, we should remove these function so it's compiler error rather than runtime error for the user.

Reply:
Yes, for that we need to remove the generic Accessor, Mutator interface all together from ValueVector and its subinterfaces. 

Since Legacy vectors are having mutator/accessor based access -- we may still need these temporary 2 functions getMutator() and getAccessor() just to comply with the interface until Legacy vectors have been removed.